File: java/src/main/java/io/aurigraph/dlt/sdk/namespace/ComplianceApi.java
package io.aurigraph.dlt.sdk.namespace;

import com.fasterxml.jackson.core.type.TypeReference;
import io.aurigraph.dlt.sdk.AurigraphClient;
import io.aurigraph.dlt.sdk.dto.AssessmentResult;
import io.aurigraph.dlt.sdk.dto.ComplianceFramework;

import java.util.List;
import java.util.Map;

/**
 * Namespace for /api/v11/compliance endpoints — regulatory compliance assessment.
 */
public class ComplianceApi {

    private final AurigraphClient client;

    public ComplianceApi(AurigraphClient client) {
        this.client = client;
    }

    /**
     * List all available compliance frameworks.
     */
    public List<ComplianceFramework> listFrameworks() {
        return client.get("/compliance/frameworks", new TypeReference<List<ComplianceFramework>>() {});
    }

    /**
     * Run a compliance assessment for an asset against a specific framework.
     */
    public AssessmentResult assess(String assetId, String framework) {
        return client.post("/compliance/assess",
                Map.of("assetId", assetId, "framework", framework), AssessmentResult.class);
    }
}

File: android/aurigraph-sdk/src/main/kotlin/io/aurigraph/sdk/apis/ComplianceApi.kt
package io.aurigraph.sdk.apis

import io.aurigraph.sdk.AurigraphClient
import io.aurigraph.sdk.models.AssessmentRequest
import io.aurigraph.sdk.models.AssessmentResult
import io.aurigraph.sdk.models.ComplianceFramework

/**
 * Regulatory compliance assessment -- frameworks, assessments, and results.
 */
class ComplianceApi(private val client: AurigraphClient) {

    /** List all available compliance frameworks. */
    suspend fun listFrameworks(): List<ComplianceFramework> =
        client.get("/compliance/frameworks")

    /** Get a specific compliance framework by ID. */
    suspend fun getFramework(frameworkId: String): ComplianceFramework =
        client.get("/compliance/frameworks/$frameworkId")

    /** Run a compliance assessment for an asset against a specific framework. */
    suspend fun assess(assetId: String, framework: String): AssessmentResult =
        client.post("/compliance/assess", AssessmentRequest(assetId, framework))

    /** Get all assessment results for an asset. */
    suspend fun getAssessments(assetId: String): List<AssessmentResult> =
        client.get("/compliance/assessments/$assetId")
}
